### DISCUSSION TITLE
Pistons

### DISCUSSION THREAD

**Discussion by larspalo** [GrandOrgue/grandorgue#139] [General]
Originally posted 2016-05-03 23:47:37.922000 by bbierbaum in GrandOrgue discussion forum on SourceForge<br><br>I am a new user of GO. I have wired some momentary buttons to use as pistons using boards from midi-hardware. For some reason, the pistons have to be pressed twice to function. This does not happen in Hauptwerk, so surely there is some setting I don&#039;t know about. Help please!

Bryan

**Comment by larspalo** [comment]
Originally posted 2016-05-04 06:48:49.086000 by [Martin Koegler](https://github.com/e9925248) in GrandOrgue discussion forum on SourceForge<br><br>Did you configure it via &quot;Detect complex midi event&quot;?
Could you please provide the settings configured via this function as well as the MIDI events generated by that piston [Audio/Midi / Log Midi Events].

**Reply by larspalo** [reply]
Originally posted 2016-05-05 02:20:02.882000 by bbierbaum in GrandOrgue discussion forum on SourceForge<br><br>I have tried configuring both ways...listen and detect complex with the same result.
The settings on this function are 9x note, channel 7, data 36, lower limit 0, upper limit 1.

The log shows channel 7, key 36, value 64. When pressed a second time the value changes to 0.

**Comment by larspalo** [comment]
Originally posted 2016-05-05 09:42:09.382000 by [Martin Koegler](https://github.com/e9925248) in GrandOrgue discussion forum on SourceForge<br><br>So your MIDI device is acting like a switch - first press = ON-Signal, second press = OFF-Signal. May I ask, what kind of GO element, you want to toggle with it?

You can configure your desired behaviour by configuring two events: Change the event type to &quot;9x Note On Toggle&quot;, then press new, use listen again and change the the event type to &quot;9x Note Off Toggle&quot;.

**Comment by larspalo** [comment]
Originally posted 2016-05-05 14:06:06.977000 by bbierbaum in GrandOrgue discussion forum on SourceForge<br><br>You are a genius! That solved my issue. Thanks for the help!

**Comment by larspalo** [comment]
Originally posted 2016-05-06 17:04:47.835000 by [Martin Koegler](https://github.com/e9925248) in GrandOrgue discussion forum on SourceForge<br><br>The next GO release will feature a &quot;on/off toogle&quot; match typ, which allows to just configure one event.

**Comment by rousseldenis** [comment]
@larspalo I will try to enhance my hardware by building new keyboard stack soon following your website instructions. I have looked around to figure out which elements to buy.

For decoder, I will take the msed32 from midi boutique. But I try to find illuminated pistons for each keyboard level. Do you have any pointer?

Many thanks

**Reply by larspalo** [reply]
Sorry, no direct idea. I used simple momentary switches without any lighting and just relied on the GO screen for visual verification.

**Reply by rousseldenis** [reply]
No problem

And which Midi encoder do you use to get input > Midi ?

**Reply by larspalo** [reply]
Nowadays I run everything I'm serious about through a ESI M4U XT USB-MIDI interface. That exact model is no longer in production, but it offers four traditional MIDI inputs that converts into one USB that GO sees as four different inputs. There are more modern MIDI interfaces that do the same thing.

**Reply by rousseldenis** [reply]
Yes, that's for the keayboards and pedalboard, I will choose the new version of ESI M4UX https://www.esi-audio.com/products/m4uex/

But my question was about thumb pistons. They need to physically get an input (on/off) to a MIDI signal.

I saw this product: https://www.midiboutique.com/midi-boutique/msed32?page=7

And I think that"s exactly what I'm looking for, but I wanted your feedback on that as there are not so much sources on the net about that.

**Reply by larspalo** [reply]
Yes, that msed32 does indeed seem to be just the thing you're looking for. From their information it should be able to drive the lighting for the pistons too. I don't have any experience with just that board, but I have ordered other things from MIDI Boutique and have only good things to say for them.

Personally, I used the two left over notes and the extra inputs from the mpc32mq pedalboard encoder for pistons. But if you want a dedicated piston encoder (for more and lighted pistons) the msed32 does indeed seem to be a very good option.
